src/pkg/logger/logger.go

```diff
@@ -239,11 +240,47 @@ func GetLogFile(logFileFlagVal string) string {
 	return r
 }
 
-func genLogger(level logLevel, format logFormat, logfile string) (*zapcore.Core, *zap.SugaredLogger) {
+// Settings records the user's preferred logging settings.
+type Settings struct {
+	File        string      // what file to log to (alt: stderr, stdout)
+	Format      logFormat   // whether to format as text (console) or json (cloud)
+	Level       logLevel    // what level to log at
+	PIIHandling piiHandling // how to obscure pii
+}
+
+// EnsureDefaults sets any non-populated settings to their default value.
+// exported for testing without circular dependencies.
+func (s Settings) EnsureDefaults() Settings {
```

Comment:
this does ensure defaults, but I'm not sure if all things are checked for validity. For example, I don't think there's a fallback for PII handling if it's outside the range of expected values. At a broader level, which functions can expect only valid input data and which functions need to guard against invalid data?

Reply:
Added checks for validity.